src/commands/deploy/deploy.js

```diff
@@ -526,6 +538,25 @@ const deploy = async (options, command) => {
   const deployFolder = await getDeployFolder({ options, config, site, siteData })
   const functionsFolder = getFunctionsFolder({ options, config, site, siteData })
   const { configPath } = site
+  const edgeFunctionsConfig = command.netlify.config.edge_functions
+
+  // build flag wasn't used and edge functions exist
+  if (!options.build && edgeFunctionsConfig && edgeFunctionsConfig.length !== 0) {
```

Comment:
@eduardoboucas should we only bundle if the build flag is absent during deploy? Netlify functions bundle all the time but I don't think it's necessary to do it a second time if the build flag is present.

Reply:
Your plan sounds good to me!